src/overlays/actors/ovl_Obj_Lupygamelift/z_obj_lupygamelift.c

```diff
@@ -28,35 +33,142 @@ const ActorInit Obj_Lupygamelift_InitVars = {
+    }
+    this->dyna.actor.home.rot.x = 0;
+    this->dyna.actor.home.rot.y = 0;
+    this->dyna.actor.home.rot.z = 0;
+    path = &globalCtx->setupPathList[OBJLUPYGAMELIFT_GET_PATH(thisx)];
+    this->unk_168 = OBJLUPYGAMELIFT_GET_7(thisx);
+    this->unk_164 = path->count;
+    if (this->unk_168 >= this->unk_164) {
+        this->unk_168 = 0;
+    }
+    this->unk_16C = Lib_SegmentedToVirtual(path->points);
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, 0x183, this->dyna.actor.world.pos.x,
+                       this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, this->dyna.actor.shape.rot.x,
+                       this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, 0);
+    if (OBJLUPYGAMELIFT_GET_C(thisx) != 0) {
+        phi_v0 = 1;
+    } else {
+        phi_v0 = 0;
+    }
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0x1D2, this->dyna.actor.home.pos.x, this->dyna.actor.home.pos.y,
+                this->dyna.actor.home.pos.z, 0, 0, 0, phi_v0);
+    func_80AF04BC(this);
+}
+
+void ObjLupygamelift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLupygamelift* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80AF0394(ObjLupygamelift* this) {
+    static f32 D_80AF0750 = 100.0f;
+    static f32 D_80AF0754 = 255.0f;
+    static f32 D_80AF0758 = 5.0f;
+    static f32 D_80AF075C = 10.0f;
+    static f32 D_80AF0760 = -240.0f;
+    static f32 D_80AF0764 = 30.0f;
+
+    f32 new_var = D_80AF0754 - D_80AF0750;
+    f32 new_var2 = D_80AF0764 - D_80AF0760;
+    f32 new_var3 = D_80AF075C - D_80AF0758;
+    f32 temp_fa0;
+    f32 phi_fa1;
+
+    temp_fa0 = this->dyna.actor.world.pos.y - D_80AF0760;
+    if (temp_fa0 < 0.0f) {
+        phi_fa1 = 0.0f;
+    } else if (new_var2 < temp_fa0) {
+        phi_fa1 = 1.0f;
+    } else {
+        phi_fa1 = temp_fa0 / new_var2;
+    }
+    this->dyna.actor.shape.shadowAlpha = D_80AF0754 - (phi_fa1 * new_var);
+    this->dyna.actor.shape.shadowScale = D_80AF075C - (phi_fa1 * new_var3);
+}
+
+void func_80AF04BC(ObjLupygamelift* this) {
+    this->unk_170 = 5;
+    this->actionFunc = func_80AF04D8;
+}
+
+void func_80AF04D8(ObjLupygamelift* this, GlobalContext* globalCtx) {
+    if (this->unk_170 == 0) {
+        func_80AF0514(this);
+    } else {
+        this->unk_170--;
+    }
+}
+
+void func_80AF0514(ObjLupygamelift* this) {
+    this->actionFunc = func_80AF0530;
+    this->dyna.actor.speedXZ = this->unk_160;
+}
+
+void func_80AF0530(ObjLupygamelift* this, GlobalContext* globalCtx) {
+    f32 step;
+    Vec3f target;
+
+    target.x = (this->unk_16C[this->unk_168].x);
+    target.y = (this->unk_16C[this->unk_168].y);
+    target.z = (this->unk_16C[this->unk_168].z);
+    step = Math_Vec3f_StepTo(&this->dyna.actor.world.pos, &target, this->dyna.actor.speedXZ);
+    if (step > 30.0f) {
+        Math_SmoothStepToF(&this->dyna.actor.speedXZ, this->unk_160, 0.5f, 5.0f, 0.1f);
```

Comment:
`unk_160` is a `targetSpeedXZ`, then (and is used as a speed in the previous function too).